pkg/cmd/backuprestore/backuprestore.go

```diff
@@ -0,0 +1,117 @@
+package backuprestore
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"io"
+	"k8s.io/klog"
+)
+
+type backupOptions struct {
+	configDir string
+	dataDir   string
+	backupDir string
+	errOut    io.Writer
+}
+
+func NewBackupCommand(errOut io.Writer) *cobra.Command {
+	backupOpts := &backupOptions{
+		errOut: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:   "cluster-backup",
+		Short: "Backs up a snapshot of etcd database and static pod resources to a given directory",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupOpts.errOut, err.Error())
+				}
+			}
+
+			must(backupOpts.Validate)
+			must(backupOpts.Run)
+		},
+	}
+	backupOpts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (r *backupOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.Set("logtostderr", "true")
+	fs.StringVar(&r.configDir, "config-dir", "/etc/kubernetes", "Path to the kubernetes config directory")
```

Comment:
Well, we need this only for getting the static pod resources. We want to store them along with the snapshot too, right?

Reply:
Yes, but where are these files on the host defined? Is it necessary to obtain them from a host mount?

Reply:
I was testing with a container of my own, but I am not still clear if we want to run this in openshift-etcd namespace or openshift-etcd-operator namespace. We need it in openshift-etcd namespace since we may want to take backup on each of the masters.